core/src/services/s3/writer.rs

```diff
@@ -159,18 +181,25 @@ impl oio::MultipartWrite for S3Writer {
 
         let status = resp.status();
 
+        let mut meta = S3Writer::parse_header_into_meta(&self.path, resp.headers())?;
+
         match status {
             StatusCode::OK => {
                 // still check if there is any error because S3 might return error for status code 200
                 // https://docs.aws.amazon.com/AmazonS3/latest/API/API_CompleteMultipartUpload.html#API_CompleteMultipartUpload_Example_4
                 let (parts, body) = resp.into_parts();
-                let maybe_error: S3Error =
-                    quick_xml::de::from_reader(body.reader()).map_err(new_xml_deserialize_error)?;
+
+                let maybe_error: S3Error = quick_xml::de::from_reader(body.clone().reader())
+                    .map_err(new_xml_deserialize_error)?;
                 if !maybe_error.code.is_empty() {
                     return Err(from_s3_error(maybe_error, parts));
                 }
 
-                Ok(())
+                let ret: CompleteMultipartUploadResult =
```

Comment:
Maybe we can embed the error struct in `CompleteMultipartUploadResult` so that we don't need to parse twice.